File: sources/server/app/models/entities/Driver.scala
package models.entities

import java.sql.Date

/**
 * Доменный класс пользователя системы
 * @param id идентификатор
 * @param pass серия-номер паспорта
 * @param license серия-номер водительского удостоверения
 * @param lastName фамилия
 * @param firstName имя
 * @param middleName отчество
 * @param phone телефон
 * @param secPhone дополнительный телефон
 * @param address адресс

 */
case class Driver(
 id: Long,
 pass: String,
 license: String,
 lastName: Option[String],
 firstName: Option[String],
 middleName: Option[String],
 phone: String,
 secPhone: String,
 comment: Option[String],
 address: String,
 creationDate: Date,
 editDate: Option[Date],
 creatorId: Option[Long],
 editorId: Option[Long]
) extends Entity

File: sources/client/app/models/driver.js
/**
 * Created by Кирилл on 25.03.2015.
 */
import DS from "ember-data";

let attr = DS.attr;
var DriversModel = DS.Model.extend({
  lastName: attr("string"),
  firstName: attr("string"),
  middleName: attr("string"),
  fio: function() {
    return this.get('lastName') + ' ' + this.get('firstName') + ' ' + this.get('middleName');
  }.property('lastName','firstName', 'middleName'),
  pass: attr("string"),
  address: attr("string"),
  license: attr("string"),
  phone: attr("string"),
  secPhone: attr("string"),
  comment: attr("string"),
  creationDate: attr("date"),
  editDate: attr("date"),
  creator: DS.belongsTo("user", {inverse: null, async: true}),
  editor: DS.belongsTo("user", {inverse: null, async: true}),
  displayName: function(){
    return this.get('fio') + ' ' + this.get('pass');
  }.property('fio','pass')
});

export default DriversModel;
